pkgs/development/libraries/p11-kit/default.nix

```diff
@@ -25,66 +24,47 @@ stdenv.mkDerivation rec {
     hash = "sha256-paLiRYgYshuedgDgW2nEsv4/Loq6qFyQMjfBJwqtHzw=";
   };
 
-  outputs = [ "out" "bin" "dev"];
+  postPatch = ''
+    substituteInPlace p11-kit/meson.build \
+      --replace 'install_dir: prefix / p11_system_config' "install_dir: '$out/etc/pkcs11'"
+  '';
+
+  outputs = [ "out" "bin" "dev" ];
+
+  strictDeps = true;
 
-  # For cross platform builds of p11-kit, libtasn1 in nativeBuildInputs
-  # provides the asn1Parser binary on the hostPlatform needed for building.
-  # at the same time, libtasn1 in buildInputs provides the libasn1 library
-  # to link against for the target platform.
-  # Hence, libtasn1 is required in both native and build inputs.
   nativeBuildInputs = [
-    autoreconfHook
-    docbook-xsl-nons
-    gtk-doc
-    installShellFiles
-    libtasn1
-    libxslt.bin
+    meson
+    ninja
     pkg-config
-    which
+    libtasn1 # asn1Parser
+    libxslt # xsltproc
+    docbook-xsl-nons
+    docbook_xml_dtd_43
+    gettext
   ];
 
   buildInputs = [
     libffi
-    libiconv
-    libintl
     libtasn1
+    libintl
   ];
 
-  autoreconfPhase = ''
-    NOCONFIGURE=1 ./autogen.sh
-  '';
-
-  configureFlags = [
-    "--enable-doc"
-    "--sysconfdir=/etc"
-    "--localstatedir=/var"
```

Comment:
It would be passed the same way to `mesonFlags` but it does not look like it is used. The only reference I found with `git log -G` is hardcoded in https://github.com/p11-glue/p11-kit/blob/d98392beb309570c228090866d77e500bc9e30db/common/runtime.c#L51

Reply:
The `p11_get_runtime_directory` function and in turn the whole codebase contains no reference to `localstatedir`, I'm pretty sure it's unused.